Templates/APIForMongoDBQuickstart-WebAPI/Services/ProductService.cs

```diff
@@ -0,0 +1,46 @@
+using MongoDB.Driver;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using APIForMongoDBQuickstart_WebAPI.Models;
+
+namespace APIForMongoDBQuickstart_WebAPI.Services
+{
+    public class ProductService : IProductService
+    {
+        private readonly IMongoCollection<Product> _products;        
+
+        public ProductService(MongoService mongo, IDatabaseSettings settings)
+        {
+            var db = mongo.GetClient().GetDatabase(settings.DatabaseName);
+            _products = db.GetCollection<Product>(settings.ProductCollectionName);
+        }
+
+        public Task<List<Product>> GetNAsync(int n)
+        {
+            return  _products.Find(product => true).Limit(n).ToListAsync();
```

Comment:
Should there be retry logic to handle transient failures?

Reply:
I believe there is some retry logic in the driver itself, but it does require the user to configure it on their account. Since this is intended to be a quickstart to learn the API for MongoDB there was some discussion about error handling maybe being too heavy, but I think you're right and I'll add more logic here

Reply:
The only reason I bring it up is we get a lot of issues with customers not realizing they need to setup retry logic. I understand it might make the quick start heavy.